pandas/tests/io/json/test_pandas.py

```diff
@@ -1665,13 +1665,21 @@ def test_json_multiindex(self, dataframe, expected):
         assert result == expected
 
     def test_to_s3(self, s3_resource):
+        import time
+
         # GH 28375
         mock_bucket_name, target_file = "pandas-test", "test.json"
         df = DataFrame({"x": [1, 2, 3], "y": [2, 4, 6]})
         df.to_json(f"s3://{mock_bucket_name}/{target_file}")
-        assert target_file in (
-            obj.key for obj in s3_resource.Bucket("pandas-test").objects.all()
-        )
+        timeout = 5
+        while True:
```

Comment:
Hmm, this is concerning. Do you know what's causing it? I would think that everything is synchronous.

Reply:
I don't know why... S3 is not supposed to be immediately consistent, maybe the botocore caches or moto doesn't update it's index immediately, or something like that.